docs/quickstart/hello-world-precompile-tutorial.md

```diff
@@ -0,0 +1,926 @@
+
+With state access, we can modify balances, read/write the storage of other contracts, and could even hook into external storage outside of the bounds of the EVM’s merkle trie (note: this would come with repercussions for state sync since part of the state would be moved off of the merkle trie). We can now write custom logic to make our own EVM.
+
+### Assumption of Knowledge
+
+We assume that the user has knowledge of git, golang, and javascript. It is not necessary to have knowlege of the EVM, but it is extremely helpful. Here are some resources to get started.
+
+- [The Ethereum Virtual Machine](https://github.com/ethereumbook/ethereumbook/blob/develop/13evm.asciidoc)
+- [Precompiles in Solidity](https://medium.com/@rbkhmrcr/precompiles-solidity-e5d29bd428c4)
+- [Deconstructing a Smart Contract](https://blog.openzeppelin.com/deconstructing-a-solidity-contract-part-i-introduction-832efd2d7737/)
+- [Layout of State Variables in Storage](https://docs.soliditylang.org/en/v0.8.10/internals/layout_in_storage.html)
+- [Layout in Memory](https://docs.soliditylang.org/en/v0.8.10/internals/layout_in_memory.html)
+- [Layout of Call Data](https://docs.soliditylang.org/en/v0.8.10/internals/layout_in_calldata.html)
+- [Contract ABI Specification](https://docs.soliditylang.org/en/v0.8.10/abi-spec.html)
+- [Precompiles in Solidity](https://medium.com/@rbkhmrcr/precompiles-solidity-e5d29bd428c4)
+- [Customizing the EVM with Stateful Precompiles](https://medium.com/avalancheavax/customizing-the-evm-with-stateful-precompiles-f44a34f39efd)
+
+### The Process
+
+We will first create a Solidity interface that our precompile will implement. Then we will use the precompile tool to autogenerate functions and fill out the rest. We're not done yet! We will then have to update a few more places within the Subnet-EVM. Some of this work involves assigning a precompile address, adding the precompile to the list of Subnet-EVM precompiles, and finally enabling the precompile. Now we can see our functions in action as we write another solidity smart contract that interacts with our precompile. Lastly, we will write some tests to make sure everything works as promised. We will also have an [official tutorial](https://github.com/ava-labs/hello-world-official-precompile-tutorial) with step by step commits you can follow to double check your work.
+
+### Prerequisites
+
+- Clone the [Subnet-EVM](https://github.com/ava-labs/subnet-evm) repo.
+- Clone [Avalanchego](https://github.com/ava-labs/avalanchego) repo.
+- Install [Avalanche Network Runner](https://docs.avax.network/subnets/network-runner)
+- Download [solc](https://docs.soliditylang.org/en/v0.8.9/installing-solidity.html)
```

Comment:
This documentation on installing solc is so confusing, I followed npm route and I still don't have solc after running `npm install -g solc`. We can/should do better in this tutorial to be very clear on how to install solc itself.

Reply:
Clairified